i2c/bmp280_i2c/bmp280_i2c.c

```diff
@@ -0,0 +1,260 @@
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ **/
+
+#include <stdio.h>
+
+#include "hardware/i2c.h"
+#include "pico/binary_info.h"
+#include "pico/stdlib.h"
+
+/* Example code to talk to a BMP280 temperature and pressure sensor
+
+   NOTE: Ensure the device is capable of being driven at 3.3v NOT 5v. The Pico
+   GPIO (and therefore I2C) cannot be used at 5v.
+
+   You will need to use a level shifter on the I2C lines if you want to run the
+   board at 5v.
+
+   Connections on Raspberry Pi Pico board, other boards may vary.
+
+   GPIO PICO_DEFAULT_I2C_SDA_PIN (on Pico this is 4 (pin 6)) -> SDA on BMP280
+   board GPIO PICO_DEFAULT_I2C_SCK_PIN (on Pico this is 5 (pin 7)) -> SCL on
+   BMP280 board 3.3v (pin 36) -> VCC on BMP280 board GND (pin 38)  -> GND on
+   BMP280 board
+*/
+
+// number of calibration registers to be read
+#define NUM_CALIB_PARAMS 24
+
+// device has default bus address of 0x76
+const uint8_t addr = 0x76;
+
+// hardware registers
+const uint8_t REG_CONFIG = 0xF5;
+const uint8_t REG_CTRL_MEAS = 0xF4;
+const uint8_t REG_RESET = 0xE0;
+
+const uint8_t REG_TEMP_XLSB = 0xFC;
+const uint8_t REG_TEMP_LSB = 0xFB;
+const uint8_t REG_TEMP_MSB = 0xFA;
+
+const uint8_t REG_PRESSURE_XLSB = 0xF9;
+const uint8_t REG_PRESSURE_LSB = 0xF8;
+const uint8_t REG_PRESSURE_MSB = 0xF7;
+
+// calibration registers
+const uint8_t REG_DIG_T1_LSB = 0x88;
+const uint8_t REG_DIG_T1_MSB = 0x89;
+const uint8_t REG_DIG_T2_LSB = 0x8A;
+const uint8_t REG_DIG_T2_MSB = 0x8B;
+const uint8_t REG_DIG_T3_LSB = 0x8C;
+const uint8_t REG_DIG_T3_MSB = 0x8D;
+const uint8_t REG_DIG_P1_LSB = 0x8E;
+const uint8_t REG_DIG_P1_MSB = 0x8F;
+const uint8_t REG_DIG_P2_LSB = 0x90;
+const uint8_t REG_DIG_P2_MSB = 0x91;
+const uint8_t REG_DIG_P3_LSB = 0x92;
+const uint8_t REG_DIG_P3_MSB = 0x93;
+const uint8_t REG_DIG_P4_LSB = 0x94;
+const uint8_t REG_DIG_P4_MSB = 0x95;
+const uint8_t REG_DIG_P5_LSB = 0x96;
+const uint8_t REG_DIG_P5_MSB = 0x97;
+const uint8_t REG_DIG_P6_LSB = 0x98;
+const uint8_t REG_DIG_P6_MSB = 0x99;
+const uint8_t REG_DIG_P7_LSB = 0x9A;
+const uint8_t REG_DIG_P7_MSB = 0x9B;
+const uint8_t REG_DIG_P8_LSB = 0x9C;
+const uint8_t REG_DIG_P8_MSB = 0x9D;
+const uint8_t REG_DIG_P9_LSB = 0x9E;
+const uint8_t REG_DIG_P9_MSB = 0x9F;
+
+const uint8_t BMP280_WRITE_MODE = 0xFE;
+const uint8_t BMP280_READ_MODE = 0xFF;
+
+struct bmp280_calib_param {
+  // temperature params
+  uint16_t dig_t1;
+  int16_t dig_t2;
+  int16_t dig_t3;
+
+  // pressure params
+  uint16_t dig_p1;
+  int16_t dig_p2;
+  int16_t dig_p3;
+  int16_t dig_p4;
+  int16_t dig_p5;
+  int16_t dig_p6;
+  int16_t dig_p7;
+  int16_t dig_p8;
+  int16_t dig_p9;
+
+  // intermediate result for pressure calc
+  int32_t t_fine;
+};
+
+#ifdef i2c_default
+void bmp280_init() {
+  // use the "handheld device dynamic" optimal setting (see datasheet)
+  uint8_t buf[2];
+
+  // 500ms sampling time, x16 filter, not set
+  uint8_t reg_config_val = ((0x04 << 5) | (0x05 << 2)) & 0xFC;
```

Comment:
Looks like this (and `reg_ctrl_meas_val`) are effectively consts?

Reply:
Changed